src/Computation/ContextExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.Z3;
+
+namespace Symbolica.Computation
+{
+    internal static class ContextExtensions
+    {
+        public static FPExpr MakeFloat(this Context self, string value, FPSort sort)
+        {
+            var (sign, exponent, significand) = Parse(value);
+
+            return self.MkFP(sign, exponent, significand, sort);
+        }
+
+        private static (bool, long, ulong) Parse(string value)
+        {
+            return value[0] == '-'
+                ? Parse(true, value[1..])
+                : Parse(false, value);
+        }
+
+        private static (bool, long, ulong) Parse(bool sign, string value)
+        {
+            var index = value.IndexOfAny(new[] {'e', 'E'});
+
+            return index == -1
+                ? Parse(sign, 0L, value)
+                : Parse(sign, long.Parse(value[(index + 1)..]), value[..index]);
+        }
+
+        private static (bool, long, ulong) Parse(bool sign, long exponent, string value)
+        {
+            var index = value.IndexOf('.');
+
+            return index == -1
+                ? (sign, exponent, ulong.Parse(value))
+                : (sign, exponent - (value.Length - 1L - index), ulong.Parse(value[..index] + value[(index + 1)..]));
+        }
+    }
+}
```

Comment:
If you can see a better way to do this nonsense I would be keen on it. I think it is probably solid but it's not the kind of thing I like to get involved with. Note that this implementation is probably actually more powerful than strictly required, because it should handle a mix of decimal and exponential notation, although I highly doubt LLVM would ever actually generate such a thing. So perhaps this reasonable assumption would allow a simpler implementation overall or the use of some dotnet API I don't know about.

Reply:
Could another approach be to instead use the built in `Parse`/`TryParse` methods and then use `Math.Sign` etc to destructure the numeric value into the data you need?

Not sure if those LLVM string representations are parseable by `System.Double.Parse` though. 

Also, this feels like it would benefit from some tests.

Reply:
It's not so much that the LLVM notations are not parseable. They are fairly standard so I think they probably would be. The issue is that these methods only exist in dotnet for IEEE 754 `binary32` and `binary64` (and maybe now `binary16`) but for any other sizes some precision would be lost (or incorrectly gained). The exact bit representation would not be maintained (which in general people don't care about, but in this we really do). This is why I can't even just extract the sign from the significand and then absolute value it. That extra bit might be needed.

You can see elsewhere in this PR that for 32-bit and 64-bit constant floats I do actually just parse them directly because `System.Single` and `System.Double` are compliant with IEEE 754.

But yes, I probably should write some tests.

Reply:
Turns out I totally misunderstood what Z3 was expecting for all this, and floats are pretty damn complicated. But this should hopefully be good to go now.